sqlglot/dialects/athena.py

```diff
@@ -1,31 +1,131 @@
+    on the query type.
+
+    As at 2024-09-10, assuming your Athena workgroup is configured to use "Athena engine version 3", the following engines exist:
+
+    Hive:
+     - Accepts mostly the same syntax as Hadoop / Hive
+     - Uses backticks to quote identifiers
+     - Has a distinctive DDL syntax (around things like setting table properties, storage locations etc) that is different from Trino
+     - Used for *most* DDL, with some exceptions that get routed to the Trino engine instead:
+        - CREATE [EXTERNAL] TABLE (without AS SELECT)
+        - ALTER
+        - DROP
+
+    Trino:
+      - Uses double quotes to quote identifiers
+      - Used for DDL operations that involve SELECT queries, eg:
+        - CREATE VIEW
+        - CREATE TABLE... AS SELECT
+      - Used for DML operations
+        - SELECT, INSERT, UPDATE, DELETE, MERGE
+
+    The SQLGlot Athena dialect tries to identify which engine a query would be routed to and then uses the parser / generator for that engine
+    rather than trying to create a universal syntax that can handle both types.
+    """
+
     class Tokenizer(Trino.Tokenizer):
+        """
+        The Tokenizer is flexible enough to tokenize queries across both the Hive and Trino engines
+        """
+
         IDENTIFIERS = ['"', "`"]
         KEYWORDS = {
+            **Hive.Tokenizer.KEYWORDS,
             **Trino.Tokenizer.KEYWORDS,
             "UNLOAD": TokenType.COMMAND,
         }
 
+    class HiveParser(Hive.Parser):
+        """
+        Parse queries for the Athena Hive execution engine
+        """
+
+        pass
+
     class Parser(Trino.Parser):
+        """
+        Parse queries for the Athena Trino execution engine
+        """
+
         STATEMENT_PARSERS = {
             **Trino.Parser.STATEMENT_PARSERS,
             TokenType.USING: lambda self: self._parse_as_command(self._prev),
         }
 
-    class Generator(Trino.Generator):
-        WITH_PROPERTIES_PREFIX = "TBLPROPERTIES"
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
 
-        PROPERTIES_LOCATION = {
-            **Trino.Generator.PROPERTIES_LOCATION,
-            exp.LocationProperty: exp.Properties.Location.POST_SCHEMA,
-        }
+            self._hive_parser = Athena.HiveParser(*args, **kwargs)
+
+        def parse(
+            self, raw_tokens: t.List[Token], sql: t.Optional[str] = None
+        ) -> t.List[t.Optional[exp.Expression]]:
+            if _parse_as_hive(raw_tokens):
+                return self._hive_parser.parse(raw_tokens, sql)
+
+            return super().parse(raw_tokens, sql)
+
+    class HiveGenerator(Hive.Generator):
+        """
+        Generating queries for the Athena Hive execution engine
+        """
+
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+
+            self._identifier_start = "`"
+            self._identifier_end = "`"
```

Comment:
Is this necessary? If I'm not mistaken, `Hive.Generator` should already initialize these attributes correctly. If we can get rid of it, then ditto about instantiating inline.

Reply:
You'd think so, however I noticed:

```
>>> q = parse_one("select * from foo")
>>> gen = Hive.Generator(identify=True)
>>> gen.generate(q)
'SELECT * FROM "foo"'
```

But you just triggered me to dig deeper into why this is and the following works as expected:
```
>>> gen = Hive.Generator(identify=True, dialect="hive")
>>> gen.generate(q)
'SELECT * FROM `foo`'
```

So i'll tidy this up, thanks!